test/e2e/util/node.go

```diff
@@ -24,10 +26,19 @@ func EnsureNodeCommunication(t *testing.T, ctx context.Context, client crclient.
 		g.Expect(err).NotTo(HaveOccurred(), "couldn't load guest kubeconfig")
 		guestClient := kubeclient.NewForConfigOrDie(guestConfig)
 
-		podList, err := guestClient.CoreV1().Pods("kube-system").List(ctx, metav1.ListOptions{LabelSelector: "app=konnectivity-agent"})
-		g.Expect(err).NotTo(HaveOccurred())
-		g.Expect(podList.Items).ToNot(BeEmpty())
-		_, err = guestClient.CoreV1().Pods("kube-system").GetLogs(podList.Items[0].Name, &corev1.PodLogOptions{Container: "konnectivity-agent"}).DoRaw(ctx)
+		err = wait.PollImmediateWithContext(ctx, 10*time.Second, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
```

Comment:
how is this change related to this PR?

Reply:
some runs failed to get the `konnectivity-agent`  pod immediately after switching from private to public. This simply adds a retry to solve that.

Reply:
Do we know what was causing the gail to get?
Can we please add a comment elaborating the above so the change is justified and we understand where this is coming from in future?